src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -286,6 +286,11 @@ static int get_xclbin_mem_topology(const char *buffer, XmaXclbinInfo *xclbin_inf
         const char *data = &buffer[ip_hdr->m_sectionOffset];
         const mem_topology *mem_topo = reinterpret_cast<const mem_topology *>(data);
         auto& xma_mem_topology = xclbin_info->mem_topology;
+        const axlf_section_header *mem_grp_hdr = xclbin::get_axlf_section(xclbin, ASK_GROUP_TOPOLOGY);
```

Comment:
This is always going to return a section.  For old pre 2020.2 xclbins it will return MEM_TOPOLOGY, for 2020.2 xclbins it will return the GROUP_TOPOLOGY section which always exists whether grouping it used or not.

Reply:
Wrong. This is from xclbin namespace and not from xrt::xclbin

Reply:
Ok, I see, but I am not entirely wrong :-)   ASK_GROUP_TOPOLOGY will always be present in 2020.2 xclbins regardless of grouping or not, so it can not really be used to determine if you should search from back or not.   My point is that you should probably always search from back.

Reply:
Also using xclbin:: directly is not super cool either, since xrt:;xclbin does a check on xrt.ini to see if any grouping (present or not) should be ignored, in which case it returns MEM_TOPOLOGY and CONNECTIVITY instead of GROUP_TOPOLOGY and GROUP_CONNECTIVITY.